packages/pluggableWidgets/popup-menu-web/src/PopupMenu.tsx

```diff
@@ -0,0 +1,141 @@
+// import { isBehindRandomElement, unBlockElement } from "@mendix/pluggable-widgets-tools";
+import {
+    isBehindElement,
+    isBehindRandomElement,
+    isElementPartiallyOffScreen,
+    isElementVisibleByUser,
+    moveAbsoluteElementOnScreen,
+    unBlockAbsoluteElement
+} from "../../../tools/pluggable-widgets-tools/src/index";
+import { ReactElement, useState, createElement, useCallback, RefObject, useEffect, useRef } from "react";
+import { executeAction } from "@widgets-resources/piw-utils";
+import { ActionValue } from "mendix";
+
+import { PopupMenuContainerProps } from "../typings/PopupMenuProps";
+import "./ui/PopupMenu.scss";
+
+export default function PopupMenu(props: PopupMenuContainerProps): ReactElement {
+    const ref = useRef<HTMLDivElement>(null);
+    useOnClickOutside(ref, () => setVisibility(false));
```

Comment:
Nice hook!

Reply:
Is it something worth extracting to a more global-level hook file?

Reply:
Yes, I've moved it to `document.ts`